pkgs/development/compilers/oraclejdk/jdk-linux-base.nix

```diff
@@ -103,7 +103,7 @@ let result = stdenv.mkDerivation rec {
       mv $sourceRoot $out
     fi
 
-    for file in $out/*
+    for file in $(ls $out/* | grep -v "src.zip")
```

Comment:
The above is an example of an inverted glob, so `!(*src.zip)` should do what you want.